src/docs/arch-row.dox

```diff
@@ -1,12 +1,89 @@
-/*! @arch_page arch-row Row Store
+/*! @arch_page arch-row-column Row Store & Column Store
 
-Row Stores are Btrees stored in WiredTiger that do not have a record id
-as key.  Thus, they implement a generalized version of a Btree, where
-the key and data can be arbitrary length.
+@section Definition
+
+A row oriented storage is a database management system where all columns of a row are stored
+together, the data is stored by row, while a column oriented system stores its data tables by
+column. WiredTiger supports both and each management system is represented internally by a certain
+type of B-Tree data structure.
+
+Keys and values in row-stores can be anything of arbitrary length, thus, row-stores implement a
+generalized version of a B-Tree. Furthermore, keys are explicitly stored. Column-stores use a 64-bit
+unsigned integer record id as keys, thus, column-stores implement a specialized version of a B-Tree
+where the key has a predictable length. In the case of column-stores, keys are implied given the
+B-Tree position.
+
+There are two types of column-stores in WiredTiger:
+- The fixed length column-store, as its name implies, the value has a fixed length, and furthermore
+the value is restricted to 1 and 8 bits in length. The bit length is specified when the column store
+is created. The fixed length column store has specialized use cases like bitmaps.
+- The variable length is a more general case which allows for values to have any length similarly to
+row-stores.
+
+@section In-memory and on-disk format
+
+Row-stores and column-stores have a specific in-memory and on-disk data structure. Please refer to
+the Data File Format section to learn more.
+
+@section Row-stores
+
+Row-stores are probably seen as the traditional database management system and they serve a general
+purpose. In WiredTiger, row-store keys are explicitly stored and they are usually duplicated within
+the same single file. In order to locate a specific record and pull together a complete row, a
```

Comment:
A row store doesn't need to pull things together - the data is stored as a contiguous block.

The column groups is a different concept. Column groups are defined to specify a schema for a table, and that schema can consist of multiple row and/or column stores internally.

The [code here|https://github.com/wiredtiger/wiredtiger/blob/develop/examples/c/ex_schema.c#L71] gives an example and [documentation here|http://source.wiredtiger.com/develop/schema.html], which should do a better job of explaining than I could.

Summarizing, each "column group" will result in a different table. Each of those tables can be either a row-store or a column store (or an LSM tree). It's possible for the value stored in each of those tables to correspond to several fields (e.g: A string, a uint64_t and another string), but those will all be stored together in the table.

Reply:
Thanks for the details, I will remove the notion of column groups here.

Reply:
See 3f6251754be978604cb3c8a3c30f3657b0ac8f8c